tools/type_extract/include/CppEmitter.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <filesystem>
+#include <fstream>
+#include <sstream>
+#include <vector>
+
+#include "slang/util/SmallVector.h"
+#include "slang/util/Util.h"
+
+namespace fs = std::filesystem;
+
+class HppFile {
+public:
+    explicit HppFile(std::string_view name, bool noSystemC) : fileName(std::string(name) + ".h") {
+        includes.emplace_back("ostream");
+        includes.emplace_back("cstddef");
+        includes.emplace_back("cstdint");
+        includes.emplace_back("string");
+        includes.emplace_back("sstream");
+        if (!noSystemC)
+            includes.emplace_back("systemc.h");
+    }
+
+    inline void add(std::string&& code) { hpp << code; }
```

Comment:
Note that the `inline` keyword for methods that are already defined inline is redundant.